File: base/src/main/java/io/quarkus/code/config/GitHubConfig.java
package io.quarkus.code.config;

import java.util.Optional;

import io.smallrye.config.ConfigMapping;
import io.smallrye.config.WithName;

@ConfigMapping(prefix = "io.quarkus.code.github")
public interface GitHubConfig {

    Optional<String> clientId();

    Optional<String> clientSecret();
}

File: api/src/main/kotlin/io/quarkus/code/config/GitHubConfig.kt
package io.quarkus.code.config

import io.smallrye.config.ConfigMapping
import io.smallrye.config.WithName
import java.util.*

@ConfigMapping(prefix = "io.quarkus.code.github")
interface GitHubConfig {

    @get:WithName("client-id")
    val clientId:  Optional<String>

    @get:WithName("client-secret")
    val clientSecret: Optional<String>
}